pkg/model/openstackmodel/servergroup.go

```diff
@@ -43,7 +43,23 @@ var _ fi.ModelBuilder = &ServerGroupModelBuilder{}
 
 // See https://specs.openstack.org/openstack/nova-specs/specs/newton/approved/lowercase-metadata-keys.html for details
 var instanceMetadataNotAllowedCharacters = regexp.MustCompile("[^a-zA-Z0-9-_:. ]")
-
+// Truncate string.
+func TruncateString(str string, length int) string {
```

Comment:
I believe this could be something more like:

```
const MAX_TAG_LENGTH_OPENSTACK = 60
```
```
func formatTag(tag string) string {
    if len(tag) <= MAX_TAG_LENGTH_OPENSTACK {
        return tag
    }
    return tag[:MAX_TAG_LENGTH_OPENSTACK ]
}
```
```
			Tags: []string{
				formatTag(fmt.Sprintf("%s=%s", openstack.TagKopsInstanceGroup, groupName)),
				formatTag(fmt.Sprintf("%s=%s", openstack.TagKopsName, portTagKopsName)),
				formatTag(fmt.Sprintf("%s=%s", openstack.TagClusterName, b.ClusterName())),
				
...
```